source/gloperate-glfw/source/Window.cpp

```diff
@@ -472,25 +471,25 @@ void Window::handleEvent(WindowEvent & event)
 
 void Window::clearEventQueue()
 {
-    while (!m_eventQueue.empty())
-    {
-        delete m_eventQueue.front();
-        m_eventQueue.pop();
-    }
+    std::queue<std::unique_ptr<WindowEvent>> empty;
+    std::swap(m_eventQueue, empty);
 }
 
 bool Window::createInternalWindow(const GLContextFormat & format, int width, int height, GLFWmonitor * /*monitor*/)
 {
     // Abort if window is already created
-    assert(nullptr == m_context);
+    assert(!m_context);
     if (m_context)
     {
         return false;
     }
 
     // Create GLFW window with OpenGL context
     GLContextFactory factory;
-    m_context = static_cast<GLContext*>(factory.createBestContext(format));
+    m_context = std::unique_ptr<GLContext>(
+                static_cast<GLContext*>(factory.createBestContext(format).release())
```

Comment:
[I added it to cppassist. ](https://github.com/chistopher/cppassist/commit/f657437d6a0d373db0e7ed0e9d31996193d8d1e1) 
Unfortunately copying the deleter makes m_context's type incompatible with the cast result. 


```
m_context = cppassist::static_unique_ptr_cast<GLContext>(factory.createBestContext(format));
```
```
/usr/include/c++/5/bits/unique_ptr.h:272: 
error: no match for ‘operator=’ (operand types are ‘std::unique_ptr<gloperate_glfw::GLContext>::deleter_type {aka std::default_delete<gloperate_glfw::GLContext>}’ and ‘std::default_delete<gloperate::AbstractGLContext>’)
```
Any advice?